metashare/api/jobs.py

```diff
@@ -0,0 +1,163 @@
+        logger.error(tb)
+        scratch_org.delete()
+        raise
+
+
+def try_to_make_branch(repository, *, new_branch, base_branch):
+    branch_name = new_branch
+    counter = 0
+    while True:
+        try:
+            latest_sha = repository.branch(base_branch).latest_sha()
+            repository.create_branch_ref(branch_name, latest_sha)
+            return branch_name
+        except UnprocessableEntity:
+            counter += 1
+            branch_name = f"{new_branch}-{counter}"
+
+
+def create_branches_on_github(*, user, repo_url, project, task, repo_root):
+    """
+    Expects to be called in the context of a local github checkout.
+    """
+    gh = login(token=user.gh_token)
+    owner, repo = extract_owner_and_repo(repo_url)
+    repository = gh.repository(owner, repo)
+
+    # Make project branch:
+    if project.branch_name:
+        project_branch_name = project.branch_name
+    else:
+        prefix = get_cumulus_prefix(
+            repo_root=repo_root,
+            repo_name="Repo Name",  # A placeholder, not used.
+            repo_url=repo_url,
+            repo_owner=owner,
+            repo_branch=repository.default_branch,
+            repo_commit=repository.branch(repository.default_branch).latest_sha(),
+        )
+        project_branch_name = f"{prefix}{slugify(project.name)}"
+        project_branch_name = try_to_make_branch(
+            repository,
+            new_branch=project_branch_name,
+            base_branch=repository.default_branch,
+        )
+        project.branch_name = project_branch_name
+
+    # Make task branch:
+    if task.branch_name:
+        task_branch_name = task.branch_name
+    else:
+        task_branch_name = try_to_make_branch(
+            repository,
+            new_branch=f"{project_branch_name}__{slugify(task.name)}",
+            base_branch=project_branch_name,
+        )
+        task.branch_name = task_branch_name
+
+    return task_branch_name
+
+
+def create_org_and_run_flow(scratch_org, *, user, repo_url, repo_branch, project_path):
+    from .models import SCRATCH_ORG_TYPES
+
+    cases = {SCRATCH_ORG_TYPES.Dev: "dev_org", SCRATCH_ORG_TYPES.QA: "qa_org"}
+
+    gh = login(token=user.gh_token)
+    owner, repo = extract_owner_and_repo(repo_url)
+    repository = gh.repository(owner, repo)
+    branch = repository.branch(repo_branch)
+    commit = branch.commit
+    latest_commit = commit.sha
+    latest_commit_url = commit.html_url
+    latest_commit_at = commit.commit.author.get("date", None)
+
+    owner, repo = extract_owner_and_repo(repo_url)
+    org_config = sf_run_flow.create_org_and_run_flow(
+        repo_owner=owner,
+        repo_name=repo,
+        repo_branch=repo_branch,
+        user=user,
+        flow_name=cases[scratch_org.org_type],
+        project_path=project_path,
+    )
+    scratch_org.url = org_config.instance_url
+    scratch_org.last_modified_at = now()
+    scratch_org.expires_at = org_config.expires
```

Comment:
This does not seem to be doing anything? Every org I create has no `expires_at` set.

Reply:
We remade it as an Org, not a ScratchOrg, and I guess that means it doesn't have an expires anymore. I'll fix it.